server.cpp

```diff
@@ -70,8 +127,27 @@ void write_response(tcp::socket& socket, const Request& request, const std::stri
     }
     else
     {
-        const std::string message = "HTTP/1.1 200 OK\r\nHost: localhost\r\n\r\n" + date;
-        boost::asio::write(socket, boost::asio::buffer(message), ignored_error);
+        std::string path;
+
+        if (!work_dir.empty())
+            path = work_dir;
+        else
+            path = ".";
+
+        DIR *dir = opendir(path.c_str());
+        if (dir == NULL)
+        {
+            error_message = "HTTP/1.1 500 Directory failed to open\r\n";
+            boost::asio::write(socket, boost::asio::buffer(error_message), ignored_error);
+        }
+        else
+        {
+            boost::asio::write(socket, boost::asio::buffer(make_message(dir, path, date)), ignored_error);
+
+            const std::string request_path_file = request.path();
+            if (request_path_file != "/")
+                write_file(socket, request_path_file, path, ignored_error);
```

Comment:
In case of serving a file, it looks as if you send the table first and then the file. You don't need the table when you sending a file.